cockatrice/src/tab_debuglog.cpp

```diff
@@ -1,35 +1,48 @@
-#include "dlg_viewlog.h"
+#include "tab_debuglog.h"
 #include "logger.h"
 
 #include <QVBoxLayout>
 #include <QPlainTextEdit>
 
-DlgViewLog::DlgViewLog(QWidget *parent)
-: QDialog(parent)
+TabDebugLog::TabDebugLog(TabSupervisor *tabSupervisor, QWidget *parent)
+: Tab(tabSupervisor, parent)
 {
     logArea = new QPlainTextEdit;
     logArea->setReadOnly(true);
     
     QVBoxLayout *mainLayout = new QVBoxLayout;
     mainLayout->addWidget(logArea);
 
-    setLayout(mainLayout);
-
-    setWindowTitle(tr("Debug Log"));
-    resize(800, 500);
+    QWidget *mainWidget = new QWidget(this);
+    mainWidget->setLayout(mainLayout);
+    setCentralWidget(mainWidget);
 
     loadInitialLogBuffer();
     connect(&Logger::getInstance(), SIGNAL(logEntryAdded(QString)), this, SLOT(logEntryAdded(QString)));    
 }
 
-void DlgViewLog::loadInitialLogBuffer()
+TabDebugLog::~TabDebugLog()
+{
+    emit debugLogClosing();
+}
+
+void TabDebugLog::closeRequest()
+{
+    deleteLater();
+}
+
+void TabDebugLog::retranslateUi()
```

Comment:
It works the other way around: when the language gets changed TabSupervisor cycle all tabs, request their title and update the caption on the tab header.
The `getTabText()` method returning the tab title is already using `tr()` to return the translated title.